x-pack/plugins/viz_editor/public/common/components/visualization_modal/visualization_modal.tsx

```diff
@@ -5,63 +5,107 @@
  */
 
 import {
-  EuiButton,
-  // @ts-ignore
-  EuiCard,
+  EuiFacetButton,
+  EuiFacetGroup,
   EuiFlexGrid,
+  EuiFlexGroup,
   EuiFlexItem,
-  EuiIcon,
   EuiModal,
   EuiModalBody,
-  EuiModalFooter,
-  EuiModalHeader,
-  EuiModalHeaderTitle,
   EuiOverlayMask,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
 } from '@elastic/eui';
-import React from 'react';
+import React, { useState } from 'react';
 import { Suggestion } from '../../../editor_plugin_registry';
+import { ExpressionRenderer } from '../../../frame/expression_renderer';
 import { VisModel } from '../../lib';
 
 interface VisualizationModalProps {
   title: string;
   onClose: () => void;
   onSelect: (newVisModel: VisModel) => void;
   suggestions: Suggestion[];
+  getInterpreter: () => Promise<{ interpreter: any }>;
+  renderersRegistry: { get: (renderer: string) => any };
 }
 
 export function VisualizationModal({
   suggestions,
   title,
   onClose,
   onSelect,
+  getInterpreter,
+  renderersRegistry,
 }: VisualizationModalProps) {
+  const suggestionCategoryMap = suggestions.reduce(
+    (categoryMap, suggestion) => ({
+      ...categoryMap,
+      [suggestion.category]: [...(categoryMap[suggestion.category] || []), suggestion],
+    }),
+    {} as {
+      [category: string]: Suggestion[];
+    }
+  );
+  const [filter, setFilter] = useState<string | undefined>(undefined);
+
   return (
     <>
       <EuiOverlayMask>
-        <EuiModal onClose={onClose} initialFocus="[name=popswitch]">
-          <EuiModalHeader>
-            <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
-          </EuiModalHeader>
-
+        <EuiModal onClose={onClose} maxWidth={false} initialFocus="[name=popswitch]">
           <EuiModalBody>
-            <EuiFlexGrid columns={3}>
-              {suggestions.map(suggestion => (
-                <EuiFlexItem key={suggestion.title}>
-                  <EuiCard
-                    key={suggestion.title}
-                    onClick={() => onSelect(suggestion.visModel)}
-                    icon={<EuiIcon size="xl" type={suggestion.iconType} />}
-                    title={suggestion.title}
-                    description=""
-                  />
-                </EuiFlexItem>
-              ))}
-            </EuiFlexGrid>
+            <EuiText>
+              <h1>{title}</h1>
+            </EuiText>
+            <EuiSpacer size="m" />
+            <EuiFlexGroup className="visualizationModal">
+              <EuiFlexItem grow={false}>
+                <EuiFacetGroup style={{ maxWidth: 200 }}>
+                  <EuiFacetButton
+                    onClick={() => {
+                      setFilter(undefined);
```

Comment:
`null` would be better than `undefined` for showing that the value is intentionally not set